datahub-web-react/src/app/identity/user/SelectRole.tsx

```diff
@@ -41,13 +39,28 @@ export default function SelectRole({ user, userRoleUrn, selectRoleOptions, refet
             );
         });
 
+    const initialRole = rolesMap.get(userRoleUrn) as DataHubRole;
+    const [currentRole, setCurrentRole] = useState<DataHubRole | undefined>(initialRole);
+    const [isViewingAssignRole, setIsViewingAssignRole] = useState(false);
+
     const onSelectRole = (roleUrn: string) => {
         const roleFromMap: DataHubRole = rolesMap.get(roleUrn) as DataHubRole;
-        setRoleToAssign(roleFromMap);
+        setCurrentRole(roleFromMap);
         setIsViewingAssignRole(true);
     };
 
-    const noRoleText = 'No Role';
+    const onCancel = () => {
+        const roleFromMap: DataHubRole = rolesMap.get(userRoleUrn) as DataHubRole;
```

Comment:
isn't this just "initialRole"? Why do we need to go back to the map here?

Reply:
nah what if someone changes the role twice? then it wouldn't be initial role

Reply:
Actually I guess this should be currentRole